client/src/components/toast.js

```diff
@@ -0,0 +1,86 @@
+import { createContext } from 'preact'
+import { useState, useContext, useCallback, useEffect } from 'preact/hooks'
+import withStyles from './jss'
+
+const ToastCtx = createContext()
+
+// Use sequential IDs
+let toastIdCounter = 0
+
+function genToastId () {
+  return toastIdCounter++
+}
+
+function Toast ({ children, remove, type, id }) {
+  const wrappedRemove = useCallback(() => remove(id), [remove, id])
+  useEffect(() => {
+    const duration = 1000 * 5
+    const timeout = setTimeout(wrappedRemove, duration)
+
+    return () => clearTimeout(timeout)
+  }, [wrappedRemove])
+
+  return (
+    <div className={`toast toast--${type}`}>
+      {children}
+      <button onClick={wrappedRemove} className='btn-close' />
+    </div>
+  )
+}
+
+const ToastContainer = withStyles({
+  container: {
+    position: 'fixed',
+    top: '1em',
+    right: '1em',
+    zIndex: 9999,
+    width: '320px'
+  }
+}, ({ classes, ...props }) => <div class={classes.container} {...props} />)
+
+export function ToastProvider ({ children }) {
+  const [toasts, setToasts] = useState([])
+
+  const remove = useCallback((id) => {
+    setToasts(toasts => toasts.filter(t => id !== t.id))
```

Comment:
Would it be possible to use a `Map` for the toasts so that we dont have to do this `O(n)` removal?

Reply:
Probably not a super big issue, converting to starter issue for someone to tackle and get more familiar with codebase